internal/brokers/manager_test.go

```diff
@@ -0,0 +1,209 @@
+			}
+			if tc.rootDir == "" {
+				tc.rootDir = rootDir
+			}
+
+			got, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(tc.rootDir))
+			if tc.wantErr {
+				require.Error(t, err, "NewManager should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "NewManager should not return an error, but did")
+
+			// Grab the list of broker names from the manager to use as golden file.
+			var brokers []string
+			for _, broker := range got.AvailableBrokers() {
+				brokers = append(brokers, broker.Name)
+			}
+
+			want := testutils.LoadWithUpdateFromGoldenYAML(t, brokers)
+			require.Equal(t, want, brokers, "NewManager should return the expected brokers, but did not")
+		})
+	}
+}
+
+func TestBrokerForSessionID(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		sessionID string
+
+		wantBroker string
+		wantErr    bool
+	}{
+		"Successfully returns expected broker":         {sessionID: "success"},
+		"Returns local broker when sessionID is empty": {wantBroker: "local"},
+
+		"Error when there is no broker for sessionID": {sessionID: "does not exist", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			b := newBrokerForTests(t)
+
+			m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+			require.NoError(t, err, "Setup: could not create manager")
+
+			if tc.wantBroker == "" {
+				tc.wantBroker = strings.ReplaceAll(t.Name(), "/", "_")
+			}
+			if tc.sessionID != "does not exist" {
+				m.SetBrokerForSessionID(tc.sessionID, &b)
+			}
+
+			got, err := m.BrokerForSessionID(tc.sessionID)
+			if tc.wantErr {
+				require.Error(t, err, "BrokerForSessionID should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "BrokerForSessionID should not return an error, but did")
+
+			require.Equal(t, tc.wantBroker, got.Name, "BrokerForSessionID should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestGetBroker(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		brokerID string
+
+		wantBroker string
+		wantErr    bool
+	}{
+		"Successfully returns expected broker": {brokerID: "local", wantBroker: "local"},
+
+		"Error when broker does not exist": {brokerID: "does not exist", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+			require.NoError(t, err, "Setup: could not create manager")
+
+			got, err := m.GetBroker(tc.brokerID)
+			if tc.wantErr {
+				require.Error(t, err, "GetBroker should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "GetBroker should not return an error, but did")
+			require.Equal(t, tc.wantBroker, got.ID, "GetBroker should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestSetDefaultBrokerForUser(t *testing.T) {
```

Comment:
ok, so as you say, we could technically add a broker that does not exists, but let’s ignore this until we change the API.